File: Strings/205.Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:

        if(len(s)!=len(t)):
            return False

        w1={}
        w2={}

        x=True
        
        for i in range(0,len(s)):
            

            if(s[i] in w1 and t[i]!=w1[s[i]]):
                x=False
                break
            if(t[i] in w2 and s[i]!=w2[t[i]]):
                 x=False
                 break
            
            w1[s[i]]=t[i]
            w2[t[i]]=s[i]

        return x

File: Strings/205. Isomorphic Strings.cpp
//https://leetcode.com/problems/isomorphic-strings/submissions/907843969/

  //TC -> O(N)
  //Sc -> O(1)

class Solution {
public:
    bool isIsomorphic(string s, string t) {
        vector<int> si(128);
        vector<int> ti(128);

        for(int i=0;i<s.length();i++)
        {
            if(si[s[i]] != ti[t[i]])
                return false;
            // increment the si and ti by 1
            si[s[i]] = i+1;
            ti[t[i]] = i+1;    
        }
        return true;
    }
};
